packager/src/main/java/io/cdap/hub/publisher/GCSPublisher.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.hub.publisher;
+
+import com.google.api.gax.paging.Page;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.hash.Hashing;
+import com.google.common.io.BaseEncoding;
+import com.google.common.io.Files;
+import com.google.common.net.MediaType;
+import io.cdap.hub.Hub;
+import io.cdap.hub.Package;
+import io.cdap.hub.SignedFile;
+import io.cdap.hub.spec.CategoryMeta;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.activation.FileTypeMap;
+import javax.activation.MimetypesFileTypeMap;
+import javax.annotation.Nullable;
+
+/**
+ * Publish packages to GCS bucket.
+ */
+public class GCSPublisher implements Publisher {
```

Comment:
Would this class benefit from unit tests at all?

Reply:
yes unit tests for this class are in progress, just trying the right way to mock the Storage

Reply:
I think someone from Ingestion made a utility class for that. Let me look for it.

Reply:
I found this old thread where they talk about it. You might have to look for the right code version to find the appropriate code. The important part apparently is:
```
  private static StorageOptions storageOptions = LocalStorageHelper.getOptions();
```

https://liveramp.slack.com/archives/CCE236DFT/p1573591855066600